operations/observability/Observability.md

```diff
@@ -0,0 +1,365 @@
+
+### Configuration file for the grok-exporter
+
+The configuration file for the grok-exporter parses the Goss test log files and creates metrics from them.
+grok-exporter version 3 configuration is used for this task and to match log expressions with regular expressions.
+
+The following is an example configuration passed to the grok-exporter to get metrics:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: example_name
+  namespace: example_namespace
+data:
+  config.yml: |-
+    global:
+      config_version: 3
+    input:
+      type: file
+      paths: 
+      - /logs/goss_tests/*
+      fail_on_missing_logfile: false
+    grok_patterns:
+      - 'REGEX [regular].*expression'
+      - 'EXAMPLE [aA-zZ-].*'
+    metrics:
+      - type: gauge
+        name: example_metric
+        help: EXAMPLE METRIC HELP
+        match: '{"log_file": "%{REGEX:logfile}", "TEST": "%{EXAMPLE:ex}"}'
+        value: '{{`{{.logfile}}`}}'
+        labels:
+          example_test: '{{`{{.ex}}`}}'
+    server:
+     port: 9144
+```
+
+### Prometheus metrics and Grafana dashboard
+
+After the preceding steps the `goss_tests` metrics are seen in Prometheus when Goss tests are run. Using these metrics,
+Grafana dashboards are created to shows the Goss tests details visually.
+
+Goss test dashboard features:
+
+- The dashboard has a dropdown for products, suites, and tests.
+- By default, all products, suites, and tests are selected for overall Goss tests result.
+- The overall product result, total number of the products, products passed, products failed, and its execution time is seen.
+- Suite results and test results are seen from the dashboard.
+- Node-wise test results are seen with failed nodes, passed nodes, and the suite/test description.
+
+![Goss test dashboard](../../img/operations/GossTestsDashboard.png "Goss Test Dashboard")
+
+## Error dashboard
+
+Error dashboards provide key glance at-a-insights about what is broken and needs attention.
+
+The automatic generation of the desired Prometheus grok-exporter configuration is based upon passing in a set of regular expression patterns to detect.
+This includes, but is not limited to, regular expression patterns to match and generate metrics on phases of manual administrator steps, as well as phases of installation scripts and automation.
+
+### Error dashboards features
+
+- Automate Grafana dashboards with error and failure message during CSM upgrade and install.
+- Create dashboard for issues comparison across multiple dimensions and clusters.
+- Monitor ConMan logs from the PIT node.
+- Parse log files with grok-exporter.
+- Create the Trend, Error, and Alert frequency dashboard.
+
+### Error dashboards
+
+Dashboards tracking different types of errors during install/upgrade.
+
+- Frequency of message about known issues dashboard
```

Comment:
What does this mean? "Frequency of message about known issues"? I see from the graphic in the PR that this apparently is the name of a dashboard, but it would be good to clarify what exactly it is showing.

Reply:
Clarify in the document? This means the dashboard shows up the count (frequency) of the known issues (can be any error or failure) that is occuring during CSM installation.

Reply:
So known issues here doesn't have the usual meaning that we use in the docs? Usually when we say "known issues" we mean "failures that exist in the product that we knew about before we shipped it", usually in the context of providing them with information on how to deal with them if they are seen. In this case, are you saying it means any failure or error that is discovered during CSM installation?

Reply:
https://github.com/Cray-HPE/docs-csm/pull/3359#discussion_r1128289942 - yes correct